kratos/utilities/mortar_utilities.h

```diff
@@ -359,6 +359,31 @@ class MortarUtilities
     }
     
     /**
+     * This function calculates the normal in a specific GP with a given shape function
+     * @param N: The shape function considered
+     * @param Geom: The geometry of condition of interest
+     */
+
+    static inline array_1d<double,3> GaussPointNormal(
```

Comment:
just one remark:

you quite like returning vectors and matrices as return values of functions. This is very clean, however it relies on the optimizer to optimize out the copy on return. 
What i am trying to say, is that from a performance point of view, it is safer to do

      Vector a
      f(a)

instead of 

      Vector a = f()

which is the reason for which we preferred the first to the second in many places in Kratos

Reply:
Ok